GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -887,6 +887,12 @@ public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> get
             Lazy<IReadOnlyList<IGitRef>> getUnfilteredRefs = new(() => (getRefs ?? capturedModule.GetRefs)(RefsFilter.NoFilter));
             _ambiguousRefs = new(() => GitRef.GetAmbiguousRefNames(getUnfilteredRefs.Value));
 
+            // Get the "main" stash commit, including the reflog selector
+            Lazy<IReadOnlyCollection<GitRevision>> getStashRevs = new(() =>
+                !AppSettings.ShowStashes
+                ? new List<GitRevision>()
```

Comment:
Does it have to be `new List<>`?

Reply:
Probably not
This is some us for something done once in a process that may take several seconds, so there is not much point in optimizing.
(I have spent a lot of time optimizing also single ms for 4.0, so performance is not unimportant but focus where it makes a difference.)

Reply:
In this case it's not even about performance. This simply looks wrong. To something which is **ReadOnly** a collection meant for manipulation is assigned. This raises questions and wastes others time while trying to comprehend the meaning of the code.
`Array.Empty<GitRevision>()` set a good tone and communicates intent clearly.

Reply:
Updating
(I do not find Array.Empty to look better in any way.)